packages/codegen/src/codegen.ts

```diff
@@ -0,0 +1,316 @@
+// Copyright 2022 Cognite AS
+import { promises as fs } from 'fs';
+import * as tmp from 'tmp-promise';
+import * as pathUtil from 'path';
+
+import { generateApi } from 'swagger-typescript-api-nextgen';
+import {
+  OpenApiDocument,
+  OpenApiSchema,
+  OpenApiOperationObject,
+  isReferenceObject,
+  OpenApiPathsObject,
+  filterPaths,
+  operationsInPath,
+  ReferenceWalker,
+} from './openapi';
+import { ConfigurationOptions } from './configuration';
+
+export type StringFilter = (str: string) => boolean;
+
+export const PassThroughFilter: StringFilter = (): boolean => true;
+
+export const ServiceNameFilter = (service: string): StringFilter => {
```

Comment:
nit: These are just functions but looks like classes or types with their capital first letter - `passThroughFilter` and `serviceNameFilter`